pipedrive/types.py

```diff
@@ -84,3 +86,43 @@ def to_primitive(self, value, context=None):
             return value['id']
 
         return value
+
+
+class PipedriveListDictStringOrStringType(BaseType):
+    """
+    Pipedrive inconsistently returns values for some fields, for example: "find" for a Person just returns a string
+    for email even if there are multiple emails associated with that Person. "detail" for a Person returns a list
+    containing a dict containing strings for emails. Because of this inconsistency we need to have a field that can
+    return the correct values even if the way they are represented changes. We are opting for the list-dict-string
+    because this allows us to better control these multiple values (email addresses, phone numbers etc.).
+    """
+    def to_native(self, value, context=None):
+        if isinstance(value, list):
+            return value
+        elif isinstance(value, basestring):
+            return [{'value': value}]
+
+    def to_primitive(self, value, context=None):
+        # Pipedrive can't handle iterable objects (dicts, lists, sets, etc.) so we throw an error if we encounter
+        # one during updating or creation. Fun fact: Pipedrive can return iterable objects, but can't receive them.
+        # So when you, for example, get a person with multiple email addresses and try to update his name Pipedrive
+        # will screw up his email addresses. In that case it's just better to update manually or specify None for
+        # the email addresses (which means it won't update them at all).
+        if not isinstance(value, basestring) and isinstance(value, Iterable):
+            raise ValueError("Pipedrive can't handle iterable objects. You can either specify one value of type base"
+                             "string which will update the first record, or specify None which won't update anything")
+        return value
+
+
+class PipedriveEmailType(PipedriveListDictStringOrStringType):
+    """
+    We're just using these because the names don't suck as much.
+    """
+    pass
+
+
+class PipedrivePhonenumberType(PipedriveListDictStringOrStringType):
+    """
+    We're just using these because the names don't suck as much.
+    """
```

Comment:
Consider making it more clear which fields this is used for and refer to `PipedriveListDictStringOrStringType` in the docstring to point out what format it returns.